aws-inventory/lambda/inventory-storage-gateway.py

```diff
@@ -0,0 +1,96 @@
+logging.getLogger('botocore').setLevel(logging.WARNING)
+logging.getLogger('boto3').setLevel(logging.WARNING)
+logging.getLogger('urllib3').setLevel(logging.WARNING)
+
+RESOURCE_PATH = "storage_gateway/gateway"
+TYPE = "AWS::ElasticLoadBalancingV2::LoadBalancer"
+
+
+def lambda_handler(event, context):
+    logger.debug("Received event: " + json.dumps(event, sort_keys=True))
+    message = json.loads(event['Records'][0]['Sns']['Message'])
+    logger.info("Received message: " + json.dumps(message, sort_keys=True))
+
+    try:
+        target_account = AWSAccount(message['account_id'])
+        for r in target_account.get_regions():
+            try:
+                discover_storage_gateways(target_account, r)
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'AccessDenied':
+                    logger.warning(f"AccessDenied attempting to discover ELBs in {target_account.account_name} ({target_account.account_id}) in region {r}: {e}")
+                    continue
+
+
+    except AntiopeAssumeRoleError as e:
+        logger.error("Unable to assume role into account {}({})".format(target_account.account_name, target_account.account_id))
+        return()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'UnauthorizedOperation':
+            logger.error("Antiope doesn't have proper permissions to this account")
+            return(event)
+        logger.critical("AWS Error getting info for {}: {}".format(message['account_id'], e))
+        capture_error(message, context, e, "ClientError for {}: {}".format(message['account_id'], e))
+        raise
+    except Exception as e:
+        logger.critical("{}\nMessage: {}\nContext: {}".format(e, message, vars(context)))
+        capture_error(message, context, e, "General Exception for {}: {}".format(message['account_id'], e))
+        raise
+
+
+def discover_storage_gateways(target_account, region):
+    '''Discover all Storage Gateways'''
+
+
+    sgw_client = target_account.get_client('storagegateway', region=region)
+    response = sgw_client.list_gateways()
+    while 'Marker' in response:  # Gotta Catch 'em all!
+        for sgws in response.get("Gateways"):
+            sgw = sgw_client.describe_gateway_information(GatewayARN=sgws.get("GatewayARN"))
+            resource_item = {
+                "awsAccountId":                 target_account.account_id,
+                "awsAccountName":               target_account.account_name,
+                "resourceType":                 TYPE,
+                "configurationItemCaptureTime": str(datetime.datetime.now()),
+                "awsRegion":                    region,
+                "configuration":                sgw,
+                "supplementaryConfiguration":   {},
+                "resourceName":                 sgw.get("GatewayName"),
+                "resourceId":                   f"{target_account.acount_id}-{region}-{sgw.get('GatewayName')}",
```

Comment:
would `response['GatewayId']` be a better `resourceId`? Ideally if a resource has an identifier like vpc-abcdef or i-abc123def we should use that.